packages/gradle-plugin/src/main/kotlin/io/realm/kotlin/gradle/RealmCompilerSubplugin.kt

```diff
@@ -16,27 +16,170 @@
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeCompilation
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinSharedNativeCompilation
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinWithJavaCompilation
+import org.jetbrains.kotlin.konan.target.Architecture
+import org.jetbrains.kotlin.konan.target.Family
+import org.jetbrains.kotlin.konan.target.KonanTarget
+import java.io.File
 
-class RealmCompilerSubplugin : KotlinCompilerPluginSupportPlugin {
+internal val gradleVersion: GradleVersion = GradleVersion.current().baseVersion
+internal val gradle70: GradleVersion = GradleVersion.version("7.0")
+internal val gradle75: GradleVersion = GradleVersion.version("7.5")
+
+class RealmCompilerSubplugin : KotlinCompilerPluginSupportPlugin, ErrorWrapper {
+
+    /**
+     * Flag indicating whether we should submit analytics data to the remote endpoint
+     */
+    private var submitAnalytics: Boolean = true
+    /**
+     * Flag indicating whether we should submit analytics data to the remote endpoint
+     */
+    private var printAnalytics: Boolean = false
+
+    /**
+     * Flag to control if an exception in analytics collection should be causing the whole build to
+     * fail or just be logged and submitted as [UNKNOWN].
+     */
+    override var failOnError: Boolean = false
+
+    private var analyticsServiceProvider: Provider<AnalyticsService>? = null
+
+    private lateinit var anonymizedBundleId: String
 
     companion object {
-        // TODO LATER Consider embedding these from the build.gradle's pluginVersion task just
+        // TODO LATER Consider embedding these from the build.gradle's versionConstants task just
         //  as with the version. But leave it for now as they should be quite stable.
         // Modules has to match ${project.group}:${project.name} to make composite build work
         const val groupId = "io.realm.kotlin"
         const val artifactId = "plugin-compiler"
         const val version = PLUGIN_VERSION
+        const val coreVersion = CORE_VERSION
+
         // The id used for passing compiler options from command line
         const val compilerPluginId = "io.realm.kotlin"
+
         // Must match io.realm.kotlin.compiler.bundleIdKey
         const val bundleIdKey = "bundleId"
+
+        // Must match io.realm.kotlin.compiler.
+        const val featureListPathKey = "featureListPath"
+    }
+
+    @Suppress("NestedBlockDepth")
+    override fun apply(target: Project) {
+        super.apply(target)
+
+        // We build the anonymized bundle id here and pass it to the compiler plugin to ensure
+        // that the metrics and sync connection parameters are aligned.
+        val bundleId = target.rootProject.name + ":" + target.name
+        anonymizedBundleId = hexStringify(sha256Hash(bundleId.toByteArray()))
+
+        printAnalytics = target.providers.environmentVariable("REALM_PRINT_ANALYTICS").safeProvider().getOrElse("false").equals("true", ignoreCase = true)
```

Comment:
Feels like this `.safeProvider().getOrElse("false").equals("true", ignoreCase = true)` could be extraced to an extension function helper?